sql/src/main/java/io/crate/jobs/PageDownstreamContext.java

```diff
@@ -53,6 +54,8 @@
     private final BitSet exhausted;
     private final ArrayList<PageResultListener> listeners = new ArrayList<>();
 
+    private final AtomicReference<KeepAliveFähnchen> keepAliveFähnchen = new AtomicReference<>(KeepAliveFähnchen.PASSIVE);
```

Comment:
can probably be just volatile as compareAndSet() or something like that isn't required?